src/libs/Pusher/pusher.js

```diff
@@ -214,6 +229,10 @@ function subscribe(channelName, eventName, eventCallback = () => {}, isChunked =
                 reject(status);
             });
         } else {
+            // we don't want to noop successEventCallback as we don't want to bind noop on each subscribe call
+            if (typeof successEventCallback === 'function') {
+                channel.bind('pusher:subscription_succeeded', successEventCallback);
+            }
```

Comment:
I'm not too sure we need/want to do this. 🤔 For now, it's probably ok to only worry about the success callback for when we call `Pusher.subscribe()` the very first time. Unless you see something I don't.

Reply:
If we are only worried about the first time then probably we should remove it. But I was trying the more generic way as somebody could miss that point while resubscribe in nested code as callbacks for events can be resubscribed in the same code. Also, I thought the `pusher:subscription_succeeded` was another event rather than a success callback.

Reply:
>as somebody could miss that point while resubscribe in nested code as callbacks for events can be resubscribed in the same code

I see what you mean. But it might be better to wait until that is a problem or something we need. This might also cause other problems e.g. if we do keep calling the code that subscribes to the channel we will bind a new callback (possibly a duplicate one) to this event multiple times.

Until now, we have never needed to register a success callback like this when first subscribing. So, I think it's safer to assume we will not want to register a success callback more than once for the first subscription to the channel.

Reply:
Yeah. for this particular issue, we should remove this, There could be one more possibility as we can call 
```
Pusher.subscribe(channel, 'pusher:subscription_succeeded',  successEventCallback, false) 
```
Instead of adding a success callback here with every subscription.

Reply:
That's an interesting point. In the future, we might want to add this ability. But for now it's maybe OK to encourage use of the success callback. Here's how I envision this working...

1. Subscribe a channel that provides some realtime stream of data
2. The channel disconnects and the realtime stream is unavailable
3. The channel reconnects and we will trigger some very specific reconnection callback to catch up the user on missed data related to the specific stream.

We should not need to subscribe more than once. If we need to do something when the channel is resubscribed we also can just do this once.

Reply:
Yeah. So we can use this in two ways. 
1. Keeping this code very specific to reconnection callback to catch up with the user on missed data related to the specific stream.
2. Second logic can be used for cases where we want to call something purposely on every connection as a kind of check or something.
I will do the first only as for this issue. The second can be used later if the need arises.